File: Backend/src/main/java/com/ssh/ecommerce/service/WishlistService.java
package com.ssh.ecommerce.service;

import com.ssh.ecommerce.dto.response.WishlistItemResponse;

import java.util.List;

public interface WishlistService {
    WishlistItemResponse addToWishlist(Long userId, Long productId);
    void removeFromWishlist(Long userId, Long productId);
    List<WishlistItemResponse> getByUser(Long userId);
}

File: Frontend/src/api/wishlistService.js
import API from './axios';

export const addToWishlist = (userId, productId) => API.post(`/wishlist/${userId}/${productId}`);
export const removeFromWishlist = (userId, productId) => API.delete(`/wishlist/${userId}/${productId}`);
export const getWishlist = (userId) => API.get(`/wishlist/${userId}`);
